ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java

```diff
@@ -493,18 +513,23 @@ public void getTableList(Context ctx) {
                   }
                 })
             .collect(Collectors.toList());
-
-    // Hive tables have lower priority, append to the end
-    if (serverCatalog instanceof MixedHiveCatalogImpl) {
-      List<String> hiveTables =
-          HiveTableUtil.getAllHiveTables(
-              ((MixedHiveCatalogImpl) serverCatalog).getHiveClient(), db);
-      Set<String> arcticTables =
-          tables.stream().map(TableMeta::getName).collect(Collectors.toSet());
-      hiveTables.stream()
-          .filter(e -> !arcticTables.contains(e))
-          .sorted(String::compareTo)
-          .forEach(e -> tables.add(new TableMeta(e, TableMeta.TableType.HIVE.toString())));
+    String catalogType = serverCatalog.getMetadata().getCatalogType();
+    if (serverCatalog instanceof MixedHiveCatalogImpl || catalogType.equals("hive")) {
+      HMSClientPool hmsClientPool = null;
+      if (serverCatalog instanceof MixedHiveCatalogImpl) {
+        hmsClientPool = ((MixedHiveCatalogImpl) serverCatalog).getHiveClient();
+      } else {
+        hmsClientPool = ((ExternalCatalog) serverCatalog).getHMSClientPool();
+      }
```

Comment:
Create HMSClient in TableController. 

`ExternalCatalog` is an abstract representation of all external catalogs, and should not contain methods like `getHMSClient`